bentoml/server/marshal_server.py

```diff
@@ -14,88 +14,109 @@
 
 import logging
 import multiprocessing
-
-from gunicorn.app.base import Application
+import psutil
 
 from bentoml import config
 from bentoml.marshal.marshal import MarshalService
 from bentoml.server.instruments import setup_prometheus_multiproc_dir
 
+if psutil.POSIX:
+    # After Python 3.8, the default start method of multiprocessing for MacOS changed to
+    # spawn, which would cause RecursionError when launching Gunicorn Application.
+    # Ref:
+    # https://docs.python.org/3/library/multiprocessing.html#contexts-and-start-methods
+    #
+    # Note: https://bugs.python.org/issue33725 claims that fork method may cause crashes
+    # on MacOS.
+    multiprocessing.set_start_method('fork')
```

Comment:
is this necessary for `gunicorn_server.py`? for the case where `--enable-microbatch` is false and there is no GunicornMarshalServer worker but only model server worker(GunicornBentoServer) in the gunicorn server?